src/inspector/processing/data-processing.service.ts

```diff
@@ -62,20 +62,20 @@ export class DataProcessingService implements OnModuleInit {
           this.logger.log(`Will not save slot [${slotToWrite}]. We already have that slot in db. Skipping...`);
           return;
         }
-        const keysIndexed = await this.nodeOperatorsContract.getAllKeysIndexed();
+        const keysIndexed = await this.registryService.getAllKeysIndexed();
         const slotTime = await this.getSlotTime(slotToWrite);
         const epoch = slotToWrite / BigInt(this.config.get('FETCH_INTERVAL_SLOTS'));
         const fetcherWriter = this.fetcherWriter(slotToWrite, epoch, stateRoot, slotNumber, slotTime, keysIndexed);
-        let lidoIDs, otherCounts;
-        if (this.latestSlotInDb == 0n) {
-          // First iteration. We should fetch general validators info firstly
+        let otherCounts;
+        let lidoIDs = await this.storage.getLidoValidatorIDs(slotToWrite);
+        if (this.latestSlotInDb == 0n || lidoIDs?.length == 0 || keysIndexed.size != lidoIDs?.length) {
+          // First iteration or new validators fetched. We should fetch general validators info firstly (id)
           const slotRes = await fetcherWriter.fetchSlotData();
           otherCounts = await fetcherWriter.writeSlotData(slotRes);
           lidoIDs = await this.storage.getLidoValidatorIDs(slotToWrite);
           const epochRes = await fetcherWriter.fetchEpochData(lidoIDs);
           await fetcherWriter.writeEpochData(lidoIDs, epochRes);
         } else {
-          lidoIDs = await this.storage.getLidoValidatorIDs(this.latestSlotInDb);
           const [slotRes, epochRes] = await Promise.all([fetcherWriter.fetchSlotData(), fetcherWriter.fetchEpochData(lidoIDs)]);
           [otherCounts] = await Promise.all([fetcherWriter.writeSlotData(slotRes), fetcherWriter.writeEpochData(lidoIDs, epochRes)]);
         }
```

Comment:
I think we can leave `lidoIDs = await this.storage.getLidoValidatorIDs(slotToWrite);`
in the `if` statement body and to place the rest right after:

```js
const [slotRes, epochRes] = await Promise.all([fetcherWriter.fetchSlotData(), fetcherWriter.fetchEpochData(lidoIDs)]);
[otherCounts] = await Promise.all([fetcherWriter.writeSlotData(slotRes), fetcherWriter.writeEpochData(lidoIDs, epochRes)]);
```

Reply:
done

Reply:
reverted